doc/src/manual/arrays.md

```diff
@@ -72,7 +72,7 @@ or a series of dimension sizes passed as a variable number of arguments.
 
 [^1]: *iid*, independently and identically distributed.
 
-The syntax `[A, B, C, ...]` constructs a 1-d array (vector) of its arguments.
+The syntax `[A, B, C, ...]` constructs a 1-d array (vector) of its arguments. If all arguments have a common promotion type then they get converted to that type. In such a situation mutable arguments are stored in the new array by reference only if they do not have to be converted.
```

Comment:
Can you hard wrap the paragraph to +- 92 chars to match the surrounding formatting?